File: MiniDOMOffice/Utenze/UtenzeCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Cursore sulla tabella delle utenze
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class UtenzeCursor 
            : minidom.Databases.DBObjectCursorPO<Utenza>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorField<int> m_IDFornitore = new DBCursorField<int>("IDFornitore");
            private DBCursorStringField m_NomeFornitore = new DBCursorStringField("NomeFornitore");
            private DBCursorField<int> m_IDCliente = new DBCursorField<int>("IDCliente");
            private DBCursorStringField m_NomeCliente = new DBCursorStringField("NomeCliente");
            private DBCursorStringField m_NumeroContratto = new DBCursorStringField("NumeroContratto");
            private DBCursorStringField m_CodiceCliente = new DBCursorStringField("CodiceCliente");
            private DBCursorStringField m_CodiceUtenza = new DBCursorStringField("CodiceUtenza");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<PeriodicitaUtenza> m_TipoPeriodicita = new DBCursorField<PeriodicitaUtenza>("TipoPeriodicita");
            private DBCursorField<int> m_IntervalloPeriodicita = new DBCursorField<int>("IntervalloPeriodicita");
            private DBCursorField<DateTime> m_DataSottoscrizione = new DBCursorField<DateTime>("DataSottoscrizione");
            private DBCursorField<DateTime> m_DataInizioPeriodo = new DBCursorField<DateTime>("DataInizioPeriodo");
            private DBCursorField<DateTime> m_DataFinePeriodo = new DBCursorField<DateTime>("DataFinePeriodo");
            private DBCursorStringField m_UnitaMisura = new DBCursorStringField("UnitaMisura");
            private DBCursorField<decimal> m_CostoUnitario = new DBCursorField<decimal>("CostoUnitario");
            private DBCursorField<decimal> m_CostiFissi = new DBCursorField<decimal>("CostiFissi");
            private DBCursorStringField m_NomeValuta = new DBCursorStringField("NomeValuta");
            private DBCursorField<UtenzaFlags> m_Flags = new DBCursorField<UtenzaFlags>("Flags");
            private DBCursorStringField m_TipoMetodoDiPagamento = new DBCursorStringField("TipoMetodoDiPagamento");
            private DBCursorField<int> m_IDMetodoDiPagamento = new DBCursorField<int>("IDMetodotoDiPagamento");
            private DBCursorStringField m_NomeMetodoDiPagamento = new DBCursorStringField("NomeMetodoDiPagamento");
            private DBCursorStringField m_TipoUtenza = new DBCursorStringField("TipoUtenza");
            private DBCursorStringField m_StimatoreBolletta = new DBCursorStringField("StimatoreBolletta");

            /// <summary>
            /// Costruttore
            /// </summary>
            public UtenzeCursor()
            {
            }

            /// <summary>
            /// StimatoreBolletta
            /// </summary>
            public DBCursorStringField StimatoreBolletta
            {
                get
                {
                    return m_StimatoreBolletta;
                }
            }

            /// <summary>
            /// Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// IDFornitore
            /// </summary>
            public DBCursorField<int> IDFornitore
            {
                get
                {
                    return m_IDFornitore;
                }
            }

            /// <summary>
            /// NomeFornitore
            /// </summary>
            public DBCursorStringField NomeFornitore
            {
                get
                {
                    return m_NomeFornitore;
                }
            }

            /// <summary>
            /// IDCliente
            /// </summary>
            public DBCursorField<int> IDCliente
            {
                get
                {
                    return m_IDCliente;
                }
            }

            /// <summary>
            /// NomeCliente
            /// </summary>
            public DBCursorStringField NomeCliente
            {
                get
                {
                    return m_NomeCliente;
                }
            }

            /// <summary>
            /// NumeroContratto
            /// </summary>
            public DBCursorStringField NumeroContratto
            {
                get
                {
                    return m_NumeroContratto;
                }
            }

            /// <summary>
            /// CodiceCliente
            /// </summary>
            public DBCursorStringField CodiceCliente
            {
                get
                {
                    return m_CodiceCliente;
                }
            }

            /// <summary>
            /// CodiceUtenza
            /// </summary>
            public DBCursorStringField CodiceUtenza
            {
                get
                {
                    return m_CodiceUtenza;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// TipoPeriodicita
            /// </summary>
            public DBCursorField<PeriodicitaUtenza> TipoPeriodicita
            {
                get
                {
                    return m_TipoPeriodicita;
                }
            }

            /// <summary>
            /// IntervalloPeriodicita
            /// </summary>
            public DBCursorField<int> IntervalloPeriodicita
            {
                get
                {
                    return m_IntervalloPeriodicita;
                }
            }

            /// <summary>
            /// DataSottoscrizione
            /// </summary>
            public DBCursorField<DateTime> DataSottoscrizione
            {
                get
                {
                    return m_DataSottoscrizione;
                }
            }

            /// <summary>
            /// DataInizioPeriodo
            /// </summary>
            public DBCursorField<DateTime> DataInizioPeriodo
            {
                get
                {
                    return m_DataInizioPeriodo;
                }
            }

            /// <summary>
            /// DataFinePeriodo
            /// </summary>
            public DBCursorField<DateTime> DataFinePeriodo
            {
                get
                {
                    return m_DataFinePeriodo;
                }
            }

            /// <summary>
            /// UnitaMisura
            /// </summary>
            public DBCursorStringField UnitaMisura
            {
                get
                {
                    return m_UnitaMisura;
                }
            }

            /// <summary>
            /// CostoUnitario
            /// </summary>
            public DBCursorField<decimal> CostoUnitario
            {
                get
                {
                    return m_CostoUnitario;
                }
            }

            /// <summary>
            /// CostiFissi
            /// </summary>
            public DBCursorField<decimal> CostiFissi
            {
                get
                {
                    return m_CostiFissi;
                }
            }

            /// <summary>
            /// NomeValuta
            /// </summary>
            public DBCursorStringField NomeValuta
            {
                get
                {
                    return m_NomeValuta;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<UtenzaFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// TipoMetodoDiPagamento
            /// </summary>
            public DBCursorStringField TipoMetodoDiPagamento
            {
                get
                {
                    return m_TipoMetodoDiPagamento;
                }
            }

            /// <summary>
            /// IDMetodoDiPagamento
            /// </summary>
            public DBCursorField<int> IDMetodoDiPagamento
            {
                get
                {
                    return m_IDMetodoDiPagamento;
                }
            }

            /// <summary>
            /// NomeMetodoDiPagamento
            /// </summary>
            public DBCursorStringField NomeMetodoDiPagamento
            {
                get
                {
                    return m_NomeMetodoDiPagamento;
                }
            }

            /// <summary>
            /// TipoUtenza
            /// </summary>
            public DBCursorStringField TipoUtenza
            {
                get
                {
                    return m_TipoUtenza;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.Utenze;
            }
        }
    }
}

File: MiniDOMOffice/Utenze/UtenzeCursor.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Class Office

    ''' <summary>
    ''' Cursore sulla tabella delle utenze
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable> _
    Public Class UtenzeCursor
        Inherits DBObjectCursorPO(Of Utenza)


        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_IDFornitore As New CCursorField(Of Integer)("IDFornitore")
        Private m_NomeFornitore As New CCursorFieldObj(Of String)("NomeFornitore")
        Private m_IDCliente As New CCursorField(Of Integer)("IDCliente")
        Private m_NomeCliente As New CCursorFieldObj(Of String)("NomeCliente")
        Private m_NumeroContratto As New CCursorFieldObj(Of String)("NumeroContratto")
        Private m_CodiceCliente As New CCursorFieldObj(Of String)("CodiceCliente")
        Private m_CodiceUtenza As New CCursorFieldObj(Of String)("CodiceUtenza")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_TipoPeriodicita As New CCursorField(Of PeriodicitaUtenza)("TipoPeriodicita")
        Private m_IntervalloPeriodicita As New CCursorField(Of Integer)("IntervalloPeriodicita")
        Private m_DataSottoscrizione As New CCursorField(Of Date)("DataSottoscrizione")
        Private m_DataInizioPeriodo As New CCursorField(Of Date)("DataInizioPeriodo")
        Private m_DataFinePeriodo As New CCursorField(Of Date)("DataFinePeriodo")
        Private m_UnitaMisura As New CCursorFieldObj(Of String)("UnitaMisura")
        Private m_CostoUnitario As New CCursorField(Of Decimal)("CostoUnitario")
        Private m_CostiFissi As New CCursorField(Of Decimal)("CostiFissi")
        Private m_NomeValuta As New CCursorFieldObj(Of String)("NomeValuta")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_TipoMetodoDiPagamento As New CCursorFieldObj(Of String)("TipoMetodoDiPagamento")
        Private m_IDMetodoDiPagamento As New CCursorField(Of Integer)("IDMetodotoDiPagamento")
        Private m_NomeMetodoDiPagamento As New CCursorFieldObj(Of String)("NomeMetodoDiPagamento")
        Private m_TipoUtenza As New CCursorFieldObj(Of String)("TipoUtenza")
        Private m_StimatoreBolletta As New CCursorFieldObj(Of String)("StimatoreBolletta")
    
        Public Sub New()

        End Sub

        Public ReadOnly Property StimatoreBolletta As CCursorFieldObj(Of String)
            Get
                Return Me.m_StimatoreBolletta
            End Get
        End Property

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property IDFornitore As CCursorField(Of Integer)
            Get
                Return Me.m_IDFornitore
            End Get
        End Property

        Public ReadOnly Property NomeFornitore As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeFornitore
            End Get
        End Property

        Public ReadOnly Property IDCliente As CCursorField(Of Integer)
            Get
                Return Me.m_IDCliente
            End Get
        End Property

        Public ReadOnly Property NomeCliente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCliente
            End Get
        End Property

        Public ReadOnly Property NumeroContratto As CCursorFieldObj(Of String)
            Get
                Return Me.m_NumeroContratto
            End Get
        End Property

        Public ReadOnly Property CodiceCliente As CCursorFieldObj(Of String)
            Get
                Return Me.m_CodiceCliente
            End Get
        End Property

        Public ReadOnly Property CodiceUtenza As CCursorFieldObj(Of String)
            Get
                Return Me.m_CodiceUtenza
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property


        Public ReadOnly Property TipoPeriodicita As CCursorField(Of PeriodicitaUtenza)
            Get
                Return Me.m_TipoPeriodicita
            End Get
        End Property

        Public ReadOnly Property IntervalloPeriodicita As CCursorField(Of Integer)
            Get
                Return Me.m_IntervalloPeriodicita
            End Get
        End Property

        Public ReadOnly Property DataSottoscrizione As CCursorField(Of Date)
            Get
                Return Me.m_DataSottoscrizione
            End Get
        End Property

        Public ReadOnly Property DataInizioPeriodo As CCursorField(Of Date)
            Get
                Return Me.m_DataInizioPeriodo
            End Get
        End Property

        Public ReadOnly Property DataFinePeriodo As CCursorField(Of Date)
            Get
                Return Me.m_DataFinePeriodo
            End Get
        End Property

        Public ReadOnly Property UnitaMisura As CCursorFieldObj(Of String)
            Get
                Return Me.m_UnitaMisura
            End Get
        End Property

        Public ReadOnly Property CostoUnitario As CCursorField(Of Decimal)
            Get
                Return Me.m_CostoUnitario
            End Get
        End Property

        Public ReadOnly Property CostiFissi As CCursorField(Of Decimal)
            Get
                Return Me.m_CostiFissi
            End Get
        End Property

        Public ReadOnly Property NomeValuta As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeValuta
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property TipoMetodoDiPagamento As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoMetodoDiPagamento
            End Get
        End Property

        Public ReadOnly Property IDMetodoDiPagamento As CCursorField(Of Integer)
            Get
                Return Me.m_IDMetodoDiPagamento
            End Get
        End Property

        Public ReadOnly Property NomeMetodoDiPagamento As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeMetodoDiPagamento
            End Get
        End Property

        Public ReadOnly Property TipoUtenza As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoUtenza
            End Get
        End Property

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function


        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeUtenze"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Office.Utenze.Module
        End Function
    End Class



End Class